XenAdmin/Commands/Controls/MigrateVMToolStripMenuItem.cs

```diff
@@ -76,60 +76,59 @@ public new Command Command
             }
         }
 
-        private class MigrateVMCommand2 : Command
+        /// <summary>
+        /// This is the command used for enabling the VM's drop-right ToolStripMenuItem "Migrate to Server >"
+        /// </summary>
+        private class MigrateVmToServerCommand : Command
         {
-            public MigrateVMCommand2()
+            public MigrateVmToServerCommand()
             {
             }
 
-            public MigrateVMCommand2(IMainWindow mainWindow, IEnumerable<SelectedItem> selection)
+            public MigrateVmToServerCommand(IMainWindow mainWindow, IEnumerable<SelectedItem> selection)
                 : base(mainWindow, selection)
             {
             }
 
             private static bool CanExecute(VM vm)
             {
-                if (vm.Connection != null && vm.Connection.IsConnected)
-                    if (Helpers.FeatureForbidden(vm.Connection, Host.RestrictIntraPoolMigrate))
+                if (vm == null || vm.Connection == null || !vm.Connection.IsConnected)
+                    return false;
+
+                if (vm.is_a_template || vm.Locked)
+                    return false;
+
+                if (Helpers.FeatureForbidden(vm.Connection, Host.RestrictIntraPoolMigrate))
                         return false;
 
-                if (vm != null && !vm.is_a_template && !vm.Locked)
-                {
-                    if(Helpers.TampaOrGreater(vm.Connection))
-                        return vm.allowed_operations != null && vm.allowed_operations.Contains(vm_operations.pool_migrate);
+                if (Helpers.TampaOrGreater(vm.Connection))
+                    return vm.allowed_operations != null && vm.allowed_operations.Contains(vm_operations.pool_migrate);
 
-                    return SelectionInVisiblePool(vm.Connection) && vm.Connection.Cache.HostCount > 1 && vm.allowed_operations != null && vm.allowed_operations.Contains(vm_operations.pool_migrate);
-                    
-                }
-                return false;
+                return SelectionInVisiblePool(vm.Connection) && vm.Connection.Cache.HostCount > 1 && vm.allowed_operations != null && vm.allowed_operations.Contains(vm_operations.pool_migrate);
             }
 
             protected override bool CanExecuteCore(SelectedItemCollection selection)
             {

```

Comment:
I don't mind if it stays like this, but if this method is being refactored it could be made a lot clearer as 

if (selection.All(...)) then return selection.Any()